data/src/test/java/org/oppia/android/data/backends/gae/api/PlatformParameterServiceTest.kt

```diff
@@ -0,0 +1,122 @@
+import androidx.test.ext.junit.runners.AndroidJUnit4
+import com.google.common.truth.Truth.assertThat
+import dagger.BindsInstance
+import dagger.Component
+import dagger.Module
+import dagger.Provides
+import org.junit.Before
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.oppia.android.data.backends.gae.NetworkModule
+import org.oppia.android.testing.network.MockPlatformParameterService
+import org.oppia.android.testing.network.RetrofitTestModule
+import org.oppia.android.testing.platformparameter.TEST_BOOLEAN_PARAM_NAME
+import org.oppia.android.testing.platformparameter.TEST_BOOLEAN_PARAM_SERVER_VALUE
+import org.oppia.android.testing.platformparameter.TEST_INTEGER_PARAM_NAME
+import org.oppia.android.testing.platformparameter.TEST_INTEGER_PARAM_SERVER_VALUE
+import org.oppia.android.testing.platformparameter.TEST_STRING_PARAM_NAME
+import org.oppia.android.testing.platformparameter.TEST_STRING_PARAM_SERVER_VALUE
+import org.robolectric.annotation.LooperMode
+import retrofit2.mock.MockRetrofit
+import javax.inject.Inject
+import javax.inject.Qualifier
+import javax.inject.Singleton
+
+/**
+ * Tests for [PlatformParameterService] retrofit instance using
+ * [org.oppia.android.testing.network.MockPlatformParameterService].
+ */
+@RunWith(AndroidJUnit4::class)
+@LooperMode(LooperMode.Mode.PAUSED)
+class PlatformParameterServiceTest {
+
+  @field:[Inject MockPlatformParameterService]
+  lateinit var mockPlatformParameterService: PlatformParameterService
+
+  @Before
+  fun setUp() {
+    setUpTestApplicationComponent()
+  }
+
+  @Test
+  fun testPlatformParameterService_getPlatformParameterUsingMockService_verifyTheResponse() {
+    val response = mockPlatformParameterService.getPlatformParametersByVersion("1").execute()
+    assertThat(response.isSuccessful).isTrue()
+
+    val responseBody = response.body()
+    assertThat(responseBody).isNotNull()
+  }
+
+  @Test
+  fun testPlatformParameterService_getPlatformParameterUsingMockService_checkForStringParam() {
+    val response = mockPlatformParameterService.getPlatformParametersByVersion("1").execute()
+    val responseBody = response.body()!!
+    assertThat(responseBody[TEST_STRING_PARAM_NAME]).isEqualTo(TEST_STRING_PARAM_SERVER_VALUE)
+  }
+
+  @Test
+  fun testPlatformParameterService_getPlatformParameterUsingMockService_checkForIntegerParam() {
+    val response = mockPlatformParameterService.getPlatformParametersByVersion("1").execute()
+    val responseBody = response.body()!!
+    assertThat(responseBody[TEST_INTEGER_PARAM_NAME]).isEqualTo(TEST_INTEGER_PARAM_SERVER_VALUE)
+  }
+
+  @Test
+  fun testPlatformParameterService_getPlatformParameterUsingMockService_checkForBooleanParam() {
+    val response = mockPlatformParameterService.getPlatformParametersByVersion("1").execute()
+    val responseBody = response.body()!!
+    assertThat(responseBody[TEST_BOOLEAN_PARAM_NAME]).isEqualTo(TEST_BOOLEAN_PARAM_SERVER_VALUE)
+  }
```

Comment:
Do we need a test for a failure case? For e.g. getting a wrong version number?

Reply:
A failed response will be handled by Work Manager and not the service itself. Also while testing we use MockRetrofit instance with 100% success rate. So I guess we don't need any failure handling tests.